eodatasets3/scripts/tostac.py

```diff
@@ -5,72 +5,218 @@
     nargs=-1,
 )
-def run(odc_metadata_files: Iterable[Path]):
+def run(
+    odc_metadata_files: Iterable[Path],
+    stac_template,
+    stac_base_url,
+    explorer_base_url,
+    validate,
+):
     for input_metadata in odc_metadata_files:
         dataset = serialise.from_path(input_metadata)
 
-        project = partial(
-            pyproj.transform,
-            pyproj.Proj(init=dataset.crs),
-            pyproj.Proj(init="epsg:4326"),
-        )
-        wgs84_geometry: BaseGeometry = transform(project, dataset.geometry)
-        item_doc = dict(
-            id=dataset.id,
-            type="Feature",
-            bbox=wgs84_geometry.bounds,
-            geometry=wgs84_geometry.__geo_interface__,
-            properties={**dataset.properties, "odc:product": dataset.product.name},
-            assets={
-                # TODO: Currently assuming no name collisions.
-                **{name: {"href": m.path} for name, m in dataset.measurements.items()},
-                **{name: {"href": m.path} for name, m in dataset.accessories.items()},
-            },
-            links=[
-                # {
-                #     "rel": "self",
-                #     "href": '?',
-                # },
-                {"rel": "odc_product", "href": dataset.product.href},
-                {
-                    "rel": "alternative",
-                    "type": "text/html",
-                    "href": f"https://explorer.dea.ga.gov.au/dataset/{dataset.id}",
-                },
-            ],
-        )
-
         name = input_metadata.stem.replace(".odc-metadata", "")
         output_path = input_metadata.with_name(f"{name}.stac-item.json")
 
+        # Load STAC template data
+        if stac_template:
+            with open(stac_template, "r") as f:
+                stac_data = json.load(f)
+                if stac_data.get("stac_version", "") not in [
+                    "1.0.0-beta.1",
+                    "1.0.0-beta.2",
+                ]:
+                    raise NotImplementedError(
+                        f"STAC version {stac_data.get('stac_version', '')} not implemented"
+                    )
+        else:
+            stac_data = {}
+
+        # Create STAC dict
+        item_doc = create_stac(
+            dataset,
+            input_metadata,
+            output_path,
+            stac_data,
+            stac_base_url,
+            explorer_base_url,
+            validate,
+        )
+
         with output_path.open("w") as f:
             json.dump(item_doc, f, indent=4, default=json_fallback)
 
-        echo(output_path)
+
+def create_stac(
+    dataset: DatasetDoc,
+    input_metadata: Path,
+    output_path: Path,
+    stac_data: dict,
+    stac_base_url: str,
+    explorer_base_url: str,
+    do_validate: bool,
+) -> dict:
+    """
+    Creates a STAC document
+    """
+
+    stac_ext = ["eo", "view", "projection"]
+    stac_ext.extend(stac_data.get("stac_extensions", []))
+
+    project = partial(
```

Comment:
We should be consistent in how we do this. Using the ODC function is what we have done a few times in the past: https://github.com/opendatacube/odc-tools/blob/develop/libs/index/odc/index/stac.py#L72

Reply:
Used ODC function for CRS transform